storage/helpers_test.go

```diff
@@ -62,30 +59,15 @@ func (s *Store) DisableRaftLogQueue(disabled bool) {
 // need their raft logs truncated and then process each of them.
 // Exposed only for testing.
 func (s *Store) ForceRaftLogScanAndProcess() {
-	// NB: This copy allows calling s.mu.RUnlock() before calling MaybeAdd, which
-	// is necessary to avoid deadlocks. Without it, a concurrent call to
-	// (*Store).mu.Lock() will deadlock the system:
-	//
-	// (*Store).mu.RLock()
-	//
-	// Start of (*Store).mu.Lock() danger zone
-	//
-	// raftLogQueue.MaybeAdd
-	// raftLogQueue.shouldQueue
-	// getTruncatableIndexes
-	// (*Store).RaftStatus
-	//
-	// End of (*Store).mu.Lock() danger zone
-	//
-	// (*Store).mu.RLock()
-	//
-	// See http://play.golang.org/p/xTJiAiRfYf
-	replicas := make([]*Replica, 0, len(s.replicas))
-	s.mu.RLock()
-	for _, r := range s.replicas {
+	// This copy allows calling s.Unlock() before calling MaybeAdd, which is
```

Comment:
this comment is no longer accurate AFAICT. there's no need for a concurrent call at all; this code will deadlock all by itself.